crates/model/src/app_id.rs

```diff
@@ -6,6 +6,9 @@ use std::{
     str::FromStr,
 };
 
+/// This allows arbitrary user data to be associated with an order. This type holds the
+/// hash of the data, while the data itself is uploaded to IPFS. The hash is signed along with the
+/// order. This allows the protocol to be extended by users in their own custom ways.
```

Comment:
I would say this rather allows **us** to extend the protocol in a secure way. Because users can put whatever data they want in there but unless we support what they are trying to do that data is meaningless to the protocol.

Reply:
Yeah, I guess that would also be possible. Maybe it's enough to document how the `AppId` works and not what it could be used for. 🤷‍♂️

Reply:
> Most commonly, the user-agent is encoded...

`user-agent` is a term with a very specific meaning (e.g. `"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:107.0) Gecko/20100101 Firefox/107.0" `). I think per default we store this at cow swap:
```
{
  "version": "0.1.0",
  "appCode": "CowSwap",
  "metadata": {
    "environment": "production"
  }
}
```

Another common use case is to store the `referrer` in case you want to participate in the referral program (not sure about the details of that one, though):
```
{
    "version": "0.1.0",
    "appCode": "Yearn",
    "metadata": {
        "referrer": {
            "version": "0.1.0",
            "address": "0xFEB4acf3df3cDEA7399794D0869ef76A6EfAff52"
        }
    }
}
```

If you want to have a look at a few `AppId` contents you can sample a few orders in the DB and inspect their `AppId` with this [tool](https://codepen.io/pacara/pen/BadZgjg).